verification_service/main.py

```diff
@@ -301,7 +301,7 @@ def process_events(
 
 
 def main():
-    #Thread(target=app.run, kwargs={"port": 8001}).start()
+    Thread(target=app.run, kwargs={"port": 8001}).start()
```

Comment:
issue: I think you need to set `daemon=True`, otherwise this will keep running even if the main thread exits (the docs around this are quite bad/misleading...). And since this is the health check, it would appear healthy even though `process_events` isn't running anymore.

Reply:
Thanks, I've added that now and seems to work!